app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -726,6 +727,37 @@ class BrowserTabViewModel @Inject constructor(
         }
     }
 
+    fun userLongPressedAutocomplete(suggestion: AutoCompleteSuggestion) {
+        when (suggestion) {
+            is AutoCompleteHistorySuggestion -> showRemoveSearchSuggestionDialog(suggestion)
+            is AutoCompleteHistorySearchSuggestion -> showRemoveSearchSuggestionDialog(suggestion)
+            else -> return
+        }
+    }
+
+    private fun showRemoveSearchSuggestionDialog(suggestion: AutoCompleteSuggestion) {
+        command.value = ShowRemoveSearchSuggestionDialog(suggestion)
+    }
+
+    fun onRemoveSearchSuggestionConfirmed(suggestion: AutoCompleteSuggestion, omnibarText: String) {
+        appCoroutineScope.launch(dispatchers.main()) {
+            withContext(dispatchers.io()) {
+                pixel.fire(AUTOCOMPLETE_RESULT_DELETED)
+                when (suggestion) {
+                    is AutoCompleteHistorySuggestion -> {
+                        history.removeHistoryEntryByUrl(suggestion.url)
```

Comment:
We should be able to pass the entire suggestion and let the implementation decide which parameter to use as a key for deletion. That way, we can also more strictly enforce the 2 types of suggestions we allow to be deleted

Reply:
I thought about doing that but `AutoCompleteSuggestion` lives in the `app` module, so to do that we would need to add that dependency to `history` module.

I also don’t think that the `history` module should know what autocomplete is, so best to keep it agnostic.